packages/odyssey-react-mui/src/labs/Drawer.tsx

```diff
@@ -0,0 +1,224 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Drawer as MuiDrawer } from "@mui/material";
+
+import {
+  useOdysseyDesignTokens,
+  DesignTokens,
+} from "../OdysseyDesignTokensContext";
+
+import { Button } from "../Button";
+import { Box } from "../Box";
+import { CloseIcon } from "../icons.generated";
+import {
+  memo,
+  ReactNode,
+  useState,
+  useEffect,
+  useRef,
+  ReactElement,
+} from "react";
+
+import styled from "@emotion/styled";
+import { useTranslation } from "react-i18next";
+import type { AllowedProps } from "../AllowedProps";
+
+export const variantValues = ["temporary", "persistent"] as const;
+
+export type DrawerProps = {
+  /**
+   * An optional Button object to be situated in the Drawerfooter. Should almost always be of variant `primary`.
+   */
+  callToActionFirstComponent?: ReactElement<typeof Button>;
```

Comment:
Just a thought but, since we are enforcing the primary action be on the right, we should probably rename these props to something more descriptive like `primaryCallToActionComponent` maybe? Any thoughts?

Reply:
good point!